src/app/pages/wallet/wallet.component.ts

```diff
@@ -12,45 +13,29 @@ import { Subscription } from 'rxjs';
     styleUrls: ['./wallet.component.scss'],
 })
 export class WalletComponent implements OnInit, OnDestroy {
-    actor: string | undefined;
-    balances: { [symbol: string]: string } = {};
-    loading = false;  // Track loading state
-    private sessionSubscription!: Subscription;
+    balances: Balance[] = [];
+    loading = false;  // Add loading state
+    private balanceSubscription!: Subscription;
 
     constructor(
-        private sessionService: SessionService,
-        private balanceService: TokenBalanceService
+        public sessionService: SessionService, // Set sessionService as public
+        private tokenBalanceService: TokenBalanceService
     ) {}
```

Comment:
It's best to keep all services private to ensure they are not used outside the component's TypeScript code. Otherwise, you may lose control over where the service is being used or whether it is being used at all.

```suggestion
        private sessionService: SessionService,
        private tokenBalanceService: TokenBalanceService
    ) {}

    get actor(): string | undefined {
        return this.sessionService.currentSession?.actor;
    }
```

Reply:
Okay, got it. Services private. Thanks!